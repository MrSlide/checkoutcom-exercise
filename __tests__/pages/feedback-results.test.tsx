import React from 'react'
import ShallowRenderer from 'react-test-renderer/shallow'

import FeedbackResultsPage from '../../src/pages/results'
import useMostRecentComments from '../../src/hooks/use-most-recent-comments'
import useRatingsCount from '../../src/hooks/use-ratings-count'

jest.mock('../../src/hooks/use-most-recent-comments')
jest.mock('../../src/hooks/use-ratings-count')

describe('FeedbackResultsPage component', function () {
  const mockProps = {
    mostRecentComments: [
      {
        comment: 'Lorem ipsum dolore sit amet',
        createdBy: 'John Doe',
        id: '123',
      }
    ],
    ratingsCount: {
      1: 42,
      2: 234,
      3: 454,
      4: 3454,
      5: 674,
    },
  }

  const mockUseMostRecentComments = jest.mocked(useMostRecentComments)
  const mockUseRatingsCount = jest.mocked(useRatingsCount)

  test('renders correctly', function () {
    mockUseMostRecentComments.mockImplementation(function (initial) {
      return {
        busy: false,
        data: initial,
        error: undefined,
      }
    })
    mockUseRatingsCount.mockImplementation(function (initial) {
      return {
        busy: false,
        data: initial,
        error: undefined,
      }
    })

    const renderer = new ShallowRenderer()

    const result = renderer.render(
      <FeedbackResultsPage {...mockProps}/>
    )

    expect(result).toMatchSnapshot()
  })

  test('renders correctly with data loading errors', function () {
    mockUseMostRecentComments.mockImplementation(function (initial) {
      return {
        busy: false,
        data: initial,
        error: new Error('Test'),
      }
    })
    mockUseRatingsCount.mockImplementation(function (initial) {
      return {
        busy: false,
        data: initial,
        error: new Error('Test'),
      }
    })

    const renderer = new ShallowRenderer()

    const result = renderer.render(
      <FeedbackResultsPage {...mockProps}/>
    )

    expect(result).toMatchSnapshot()
  })
})
