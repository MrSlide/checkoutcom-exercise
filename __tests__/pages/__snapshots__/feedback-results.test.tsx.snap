// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeedbackResultsPage component renders correctly 1`] = `
<main>
  <Section>
    <TitleWithAction
      action={
        <RouterLink
          href="/"
        >
          Go back
        </RouterLink>
      }
    >
      <Title>
        Feedback Results
      </Title>
    </TitleWithAction>
    <RatingsChart
      data={
        {
          "1": 42,
          "2": 234,
          "3": 454,
          "4": 3454,
          "5": 674,
        }
      }
    />
  </Section>
  <Section
    off={true}
  >
    <SubTitle>
      Latest Comments
    </SubTitle>
    <CommentList>
      <Comment
        author="John Doe"
        comment="Lorem ipsum dolore sit amet"
      />
    </CommentList>
  </Section>
</main>
`;

exports[`FeedbackResultsPage component renders correctly with data loading errors 1`] = `
<main>
  <Section>
    <TitleWithAction
      action={
        <RouterLink
          href="/"
        >
          Go back
        </RouterLink>
      }
    >
      <Title>
        Feedback Results
      </Title>
    </TitleWithAction>
    <Paragraph>
      It was not possible to load the ratings.
    </Paragraph>
  </Section>
  <Section
    off={true}
  >
    <SubTitle>
      Latest Comments
    </SubTitle>
    <Paragraph>
      It was not possible to load the latest comments.
    </Paragraph>
  </Section>
</main>
`;
